File: updater/src/main/java/updater/identifiers/WorldMapIcon1.java
package updater.identifiers;

import updater.utility.AbstractIdentifier;
import updater.utility.ClassWrapper;
import updater.utility.DependsOn;

@DependsOn(AbstractWorldMapIcon.class)
public class WorldMapIcon1 extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.isSuperClassEquals("AbstractWorldMapIcon") &&
                classNode.getInstanceFields().size() == 4;
    }
}

File: updater-runestar/src/main/java/org/runestar/client/updater/mapper/identifiers/classes/WorldMapIcon1.kt
package org.runestar.client.updater.mapper.identifiers.classes

import org.runestar.client.updater.mapper.abstractclasses.IdentityMapper
import org.runestar.client.updater.mapper.annotations.DependsOn
import org.runestar.client.updater.mapper.predicateutilities.and
import org.runestar.client.updater.mapper.predicateutilities.predicateOf
import org.runestar.client.updater.mapper.wrappers.Class2
import org.runestar.client.updater.mapper.wrappers.Field2

@DependsOn(AbstractWorldMapIcon::class)
class WorldMapIcon1 : IdentityMapper.Class() {

    override val predicate = predicateOf<Class2> { it.superType == type<AbstractWorldMapIcon>() }
            .and { it.instanceFields.size == 4 }

    @DependsOn(WorldMapLabel::class)
    class label0 : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<Field2> { it.type == type<WorldMapLabel>() }
    }
}